includes/cub.h

```diff
@@ -48,6 +49,50 @@ typedef struct s_pair_i {
 	int	y;
 }				t_pair_i;
 
+typedef struct s_window {
+	void	*mlx_ptr;
+	void	*window_ptr;
+	int		height;
+	int		width;
+}	t_window;
+
+typedef struct s_img {
+	t_window	*win;
+	void		*img_ptr;
+	char		*addr;
+	int			height;
+	int			width;
+	int			bits_per_pixel;
+	int			line_length;
+	int			endian;
+}	t_img;
+
+typedef struct s_walls {
```

Comment:
or we could have a single s_wall struct with a string and an image, and in the cub struct we have the 4 wall structs for each wall (just for increased reusability)

Reply:
Also, if we don't use the fd's, let's just remove them from the struct. In the parsing we can just close the file after we checked it's ok, and store the name of the file instead of the fd

Reply:
1. I changed this to a `t_wall` struct, that has `path` and `t_img` in it. Do please mind we know have a fixed array of size [4] inside of `*cub`:
    0: NO
    1: SO
    2: WE
    3: EA

A good way to remember this would be to make a christian cross on yourself 🤔 

2. I closed the fd at the end of the `texture_is_valid` function.
3. I initialized the path string to NULL in init_cub().
4. I freed the strdups in free_memory
5. Edited the function that checks if a function has already been defined to check if the path_name has already been set

Reply:
A christian cross 😂 I think the most readable option would be to have an enum for cardinal points where NO = 0
Honestly I think the array makes it a bit unreadable and we don't save that much space with that, so I'd go back to having 4 separate variables